fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java

```diff
@@ -415,6 +416,25 @@ List<CompactionRecord> getHistory() {
         return builder.build();
     }
 
+    @NotNull
+    public void cancelCompaction(long txnId) {
+        for (Iterator<Map.Entry<PartitionIdentifier, CompactionJob>> iterator = runningCompactions.entrySet().iterator();
+                iterator.hasNext(); ) {
+            Map.Entry<PartitionIdentifier, CompactionJob> entry = iterator.next();
+            CompactionJob job = entry.getValue();
+
+            if (job.getTxnId() == txnId) {
+                job.abort();
+                PartitionIdentifier partitionIdentifier = entry.getKey();
+                Database db = stateMgr.getDb(partitionIdentifier.getDbId());
+                if (db == null) {
+                    return;
+                }
+                abortTransactionIgnoreError(db.getId(), txnId, "user cancel compaction");
```

Comment:
```suggestion
```
Try to keep the logic of abort transactions in one place, there is no need to abort transactions here